cmd/peek-submission/main_test.go

```diff
@@ -60,20 +59,15 @@ func TestPeekBase(test *testing.T) {
 			continue
 		}
 
-		var response core.APIResponse
-		util.MustJSONFromString(stdout, &response)
-
-		if response.Locator != testCase.expectedLocator {
-			test.Errorf("Case %d: Unexpected locator code. Expected: '%s', Actual: '%s'.", i, testCase.expectedLocator, response.Locator)
-			continue
-		}
-
 		if testCase.expectedExitCode != 0 {
```

Comment:
You are not checking any output (aside from the exit status) when an error occurs.
You should be checking stdout (which should contain an error message/string).
The output may not be JSON, but you can still check it.